src/app/clusters/bindings/BindingManager.cpp

```diff
@@ -169,7 +169,7 @@ void BindingManager::FabricRemoved(FabricIndex fabricIndex)
 CHIP_ERROR BindingManager::NotifyBoundClusterChanged(EndpointId endpoint, ClusterId cluster, void * context)
 {
     VerifyOrReturnError(mInitParams.mFabricTable != nullptr, CHIP_ERROR_INCORRECT_STATE);
-    VerifyOrReturnError(mBoundDeviceChangedHandler, CHIP_NO_ERROR);
+    VerifyOrReturnError(mBoundDeviceChangedHandler != nullptr, CHIP_ERROR_INCORRECT_STATE);
```

Comment:
I think it might have been intentional by the author to just skip the processing if there's no handler register. Does it cause any problem or is against the spec?

Reply:
Currently, the caller cannot discriminate between:
1. The handler is not registered -> no processing done
2. Checks passed -> appropriate processing done

Reply:
This seems intentional to me as well: #16553 has this delta and only few others.

@gjc13 ?

Reply:
From what I can tell, most invocations of NotifyBoundClusterChanged do NOT actually do error checking :(

Only examples/light-switch-app/nrfconnect/main/BindingHandler.cpp does so.

So I believe either treating as an error as as success is OK, I also agree that being able to differentiate is reasonable.

However please do not use `INCORRECT_STATE` as that may be fabric table is null or something else. We need something specific so that callers can choose if to treat all errors as bad or single-case the `no bound handlers` as still ok.

Reply:
Maybe `CHIP_ERROR_NOT_FOUND` somewhat resembles a `binding not found` ... however for maintainability I thing we should create a new `HANDLER_NOT_FOUND` or `CHIP_ERROR_HANDLER_NOT_SET` error.